### PR TITLE
Conp develop

### PR DESCRIPTION
**Pull request by ludwig-ahrens**
**Summary**
Speed up update of mpos in fix_charge_update and compute_conp_vector. 
Added a bit more logging to identify bottlenecks.

**Related Issue(s)**

**Author(s)**

**Licensing**

By submitting this pull request, I agree, that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

**Backward Compatibility**

**Implementation Notes**

**Post Submission Checklist**

- [ ] The feature or features in this pull request is complete
- [ ] Licensing information is complete
- [ ] Corresponding author information is complete
- [ ] The source code follows the LAMMPS formatting guidelines
- [ ] Suitable new documentation files and/or updates to the existing docs are included
- [ ] The added/updated documentation is integrated and tested with the documentation build system
- [ ] The feature has been verified to work with the conventional build system
- [ ] The feature has been verified to work with the CMake based build system
- [ ] Suitable tests have been added to the unittest tree.
- [ ] A package specific README file has been included or updated
- [ ] One or more example input decks are included

**Further Information, Files, and Links**

### CODE REVIEW

File: src/USER-CONP/compute_conp_vector.cpp [hunk 1]
```diff
@@ -303,30 +329,41 @@ void ComputeConpVector::create_taglist() {
...
+  for (size_t i = 0; i < taglist.size(); i++) {
+    tag_to_iele[taglist[i]] = i;
+  }
 }
 
 /* ---------------------------------------------------------------------- */
 
 void ComputeConpVector::update_mpos() {
-  int const nmax = atom->nmax;
+  MPI_Barrier(world);
```
Comment (robeme) [inline src/USER-CONP/compute_conp_vector.cpp:349]: Was machen die MPI_Barrier hier genau?

File: src/USER-CONP/fix_charge_update.h [hunk 2]
```diff
@@ -44,9 +44,11 @@ class FixChargeUpdate : public Fix {
   std::vector<std::vector<double> > elastance;
   bigint ngroup;
   std::vector<tagint> taglist, taglist_bygroup, group_idx;
+  std::vector<int> tag_to_iele;
```
Comment (robeme) [inline src/USER-CONP/fix_charge_update.h:47]: Was macht das tag_to_iele genau bzw. was steht darin?

File: src/USER-CONP/pppm_conp.cpp [hunk 3]
```diff
@@ -861,7 +676,7 @@ void PPPMConp::start_compute() {
     if (atom->nmax > nmax) {
       memory->destroy(part2grid);
       nmax = atom->nmax;
-      memory->create(part2grid, nmax, 3, "pppm/conp:part2grid");
+      memory->create(part2grid, nmax, 3, "pppm:part2grid");
```
Comment (robeme) [inline src/USER-CONP/pppm_conp.cpp]: Hier muss das array part2grid schon noch pppm/conp:part2grid heißen

File: src/USER-CONP/fix_charge_update.h (hunk 2 above)
Comment (ludwig-ahrens) [reply src/USER-CONP/fix_charge_update.h:47]: Index in mpos für jedes tag. Dadurch brauchen wir keine verschachtelte Schleife mehr um mpos zu aktualisieren.

File: src/USER-CONP/compute_conp_vector.cpp (hunk 1 above)
Comment (ludwig-ahrens) [reply src/USER-CONP/compute_conp_vector.cpp:349]: Das wird für die Timings in den Logs gebraucht. Können wir später rausnehmen.

**Pull request merged**